scopes/dependencies/dependency-resolver/dependency-resolver.docs.md

```diff
@@ -1,235 +1,322 @@
 ---
-description: Dependency management for Bit components.
-labels: ['dependency', 'dependencies', 'component', 'bit']
+description: Simplifies the dependency management of components in a Bit workspace.
+labels: ['core aspect', 'dependencies']
 ---
 
-This extension is responsible for:
-1. detecting dependencies of components by static code analysis
-2. apply dependencies polices
-3. resolve dependencies versions
-4. calculate the final dependencies of a given component
-
-## Usage
-
-### workspace config
-The extension's workspace configuration will have the following fields:
-* policy
-* packageManager
-* strictPeerDependencies
-* packageManagerArgs
-Here is a full example of workspace config:
-(For complete reference see DependencyResolverWorkspaceConfig type)
-```js
-/**
-  * main configuration for component dependency resolution.
-  */
+The Dependency Resolver auto-generates the dependency graph for components in a Bit workspace to spare us the tedious work of manually configuring it ourselves.
+It does so by integrating dependency configurations set by various extensions (primarily the workspace) with the component's parsed `import`/`require` statements.
+In addition to that, the Dependency Resolver offers an efficient API to manually modify the generated graph using policies that can be applied on groups of components.
+
+#### Features
+
+- **Auto-generated dependency graph**:
+  The Dependency Resolver saves us time and effort by generating the dependency graph for each component in the workspace. It does so by parsing out all `import` \ `require` statements in the component's files.
+  It then determines if they are packages, components or internal implementation files.
+  If the dependencies are external packages or components, it goes on to determine their version and the dependency-type (`dependencies`, `devDependencies`, `peerDependencies`).
```

Comment:
Add that it's also calculate the type - component/package
Change the name (type) of dep/dev/peer to lifecycle type